Assets/Scripts/Layers/PolygonSelectionCalculator.cs

```diff
@@ -58,7 +58,7 @@ private void ProcessClick()
 
             var camera = Camera.main;
             Plane[] frustumPlanes = GeometryUtility.CalculateFrustumPlanes(camera);
-            var worldPoint = opticalRaycaster.GetWorldPointAtCameraScreenPoint(camera, Pointer.current.position.ReadValue());
+            var worldPoint = opticalRaycaster.WorldPoint;
```

Comment:
I understand that you want to reduce the amount of calls - but now you do it every update while that may not be necessary. Wouldn't it be better to cache the WorldPoint inside the opticalRaycaster and clear the cache every frame or so?

Also: the OpticalRaycaster is now coupled to the UI system by introducing the pointer value in the optical raycaster?

Reply:
The point is that it was being called in several scripts anyway, so this is more optimal. It is only not optimal if you never use the Optical raycaster, but then you can just not have it in the scene and it will not render every frame anyway. this does couple it to the pointer yes, but this is a tradeoff also to make it easier to use, since the reason it was broken in the first place was that the GetWorldPointAtCameraScreenPoint function was not being used correctly.
I can make a separate script that does the update call for the current pointer position, but then the scripts should be coupled to this separate component instead of the optical raycaster if that is better